pkg/controller/controller.go

```diff
@@ -47,19 +49,30 @@ type Cluster struct {
 
 // Controller object
 type Controller struct {
-	k8sclient *k8sops.K8sops
-	clusters  map[string]Cluster
-	m3admin   m3admin.M3admin
+	logger          *zap.Logger
+	k8sclient       *k8sops.K8sops
+	clusters        map[string]Cluster
+	placementClient m3admin.Placement
+	namespaceClient m3admin.Namespace
 }
 
 // New creates new instance of Reconcile
-func New(kclient *k8sops.K8sops) (*Controller, error) {
+func New(logger *zap.Logger, kclient *k8sops.K8sops) (*Controller, error) {
 	p := &Controller{
+		logger:    logger,
 		k8sclient: kclient,
 		clusters:  make(map[string]Cluster),
-		m3admin:   m3admin.New(),
 	}
-
+	placementClient, err := placement.New(logger)
+	if err != nil {
+		return nil, err
+	}
+	namespaceClient, err := namespace.New(logger)
+	if err != nil {
+		return nil, err
+	}
+	p.namespaceClient = namespaceClient
```

Comment:
what is the advantage of assigning this variable after the error checking, as opposed to assigning `namespace.New(logger)` to `p.namespaceClient` (and same with `p.placementClient`) right away? 

should we decide to keep the error checking & struct assigning separate, i think it would be helpful to keep a uniform order (namespace-placement-namespace-placement) instead of the arbitrary placement-namespace-namespace-placement order.